### PR TITLE
Add check for negative `buffer_size` inside `StreamPeerGZIP::start_(de)compression`

### LINKED ISSUE #87399: `StreamPeerGZIP.start_compression` randomly freezes Godot
### Tested versions

4.3.dev.custom_build. 0bcc0e92b

### System information

Ubuntu 22.04 CI

### Issue description

When executing (this code was automatically minimized, so it is possible, that an even more "minimal" project can be created)
```
extends Node
func _process(delta):
	var temp_variable4024 = StreamPeerGZIP.new()
	temp_variable4024.start_compression(false, -55)
	temp_variable4024.finish()
```
```
extends Node
func _process(delta):
	var temp_variable1409 = StreamPeerGZIP.new()
	temp_variable1409.start_compression(false, -82)
	temp_variable1409.put_string("")
```
Godot freezes(timeout is set to 60 seconds):
```
Godot Engine v4.3.dev.custom_build.0bcc0e92b - https://godotengine.org
modules/gdscript/gdscript_vm.cpp:719:13: runtime error: store to misaligned address 0x616001253e34 for type '<unknown> *', which requires 8 byte alignment
0x616001253e34: note: pointer points here
  01 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  0a 00 00 00 02 00 00 00  09 00 00 00 17 00 00 00
              ^ 
modules/gdscript/gdscript_vm.cpp:726:42: runtime error: load of misaligned address 0x616001253e34 for type '<unknown> *', which requires 8 byte alignment
0x616001253e34: note: pointer points here
  01 00 00 00 b2 b6 1a 53  49 56 00 00 00 00 00 00  0a 00 00 00 02 00 00 00  09 00 00 00 17 00 00 00
              ^ 
######################## Ending test ########################
core/templates/ring_buffer.h:199:7: runtime error: signed integer overflow: -2147483648 - 1 cannot be represented in type 'int'
core/templates/ring_buffer.h:179:20: runtime error: signed integer overflow: -2147483648 - 1 cannot be represented in type 'int'
core/templates/ring_buffer.h:165:13: runtime error: signed integer overflow: 20 - -2147483648 cannot be represented in type 'int'
timeout: sending signal TERM to command ‘xvfb-run’
```

The problem is that I cannot reproduce this locally, but only in CI

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

Memory leaks or asan backtraces are visible when using Godot build with sanitizers support - https://github.com/qarmin/GodotBuilds/actions (linux -> linux-editor-sanitizers)

### Steps to reproduce

Above

### Minimal reproduction project (MRP)

Above

### PR DESCRIPTION
**Pull request by nfrmtk**
Fixes #87399

### CODE REVIEW

File: core/io/stream_peer_gzip.cpp [hunk 1]
```diff
@@ -76,6 +76,7 @@ Error StreamPeerGZIP::start_decompression(bool p_is_deflate, int buffer_size) {
 
 Error StreamPeerGZIP::_start(bool p_compress, bool p_is_deflate, int buffer_size) {
 	ERR_FAIL_COND_V(ctx != nullptr, ERR_ALREADY_IN_USE);
+	ERR_FAIL_COND_V_MSG(buffer_size <= 0, ERR_INVALID_PARAMETER, "provided buffer_size is invalid. try setting buffer_size > 0");
```
Comment (AThousandShips) [inline core/io/stream_peer_gzip.cpp:79]:
```suggestion
	ERR_FAIL_COND_V(buffer_size <= 0, ERR_INVALID_PARAMETER);
```
Message is clear enough IMO, but if it should be kept it should be:
```suggestion
	ERR_FAIL_COND_V_MSG(buffer_size <= 0, ERR_INVALID_PARAMETER, "Provided buffer_size is invalid. Try setting buffer_size > 0.");
```
Or similar

**Comment by akien-mga** [comment]
Could you squash the commits? See [PR workflow](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase) for instructions.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**